### PR TITLE
Source Generator

### PR DESCRIPTION
**Pull request by madeyoga**
I have implemented a source generator to create `IRazorSliceProxy` implementations. I have also added the generator project reference to `RazorSlices.Samples.WebApp.csproj` and `RazorSlices.Samples.RazorClassLibrary.csproj`.

However, I'm encountering an error specifically with the `/lorem-stream` endpoint in the samples project.

```
Error code CS9144
Cannot intercept method 'EndpointRouteBuilderExtensions.MapGet(IEndpointRouteBuilder, string, RequestDelegate)' 
with interceptor 
'GeneratedRouteBuilderExtensionsCore.MapGet4(IEndpointRouteBuilder, string, Delegate)' 
because the signatures do not match.	
RazorSlices.Samples.WebApp (net7.0)	
G:\RazorSlices\samples\RazorSlices.Samples.WebApp\Microsoft.AspNetCore.Http.RequestDelegateGenerator\Microsoft.AspNetCore.Http.RequestDelegateGenerator.RequestDelegateGenerator\GeneratedRouteBuilderExtensions.g.cs
```

### CODE REVIEW

File: src/RazorSlices.SourceGenerator/RazorSliceProxyGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        codeBuilder.AppendLine("using System.Diagnostics.CodeAnalysis;");
+        codeBuilder.AppendLine("#nullable enable");
+        codeBuilder.AppendLine($"namespace {assemblyName}.Slices;");
+
+        foreach (var file in texts)
+        {
+            var filename = Path.GetFileNameWithoutExtension(file.Path);
+
+            codeBuilder.AppendLine($$"""
+                public sealed class {{filename}} : IRazorSliceProxy
```
Comment (DamianEdwards) [inline src/RazorSlices.SourceGenerator/RazorSliceProxyGenerator.cs:48]: The `filename` will need to be sanitized to ensure a valid class name symbol is emitted here.

File: src/RazorSlices.SourceGenerator/RazorSliceProxyGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                {
+                    [DynamicDependency(DynamicallyAccessedMemberTypes.All, TypeName, "{{assemblyName}}")]
+                    private const string TypeName = "AspNetCoreGeneratedDocument.Slices_{{filename}}, {{assemblyName}}";
```
Comment (DamianEdwards) [inline src/RazorSlices.SourceGenerator/RazorSliceProxyGenerator.cs:51]: `filename` will need to be sanitized to ensure a valid type name is emitted here.

**Comment by madeyoga** [comment]
I have added type name validation to the generator. The validation ensures the type name adheres to the following rules:
- String must start with a letter or underscore.
- String may only contain letters, digits, connector punctuation.
- String is not a reserved keyword.

If the filename contain an invalid type name, the current code will attempt to fix it.

I'm unsure whether I should fix the type name in this case.

**Comment by DamianEdwards** [comment]
This is great, thank you! I'm going to merge this into the aot branch so we can continue to refine it. Thanks again!

**Pull request merged**